crates/view/src/service.rs

```diff
@@ -107,17 +115,30 @@ impl ViewService {
         let (worker, sct, error_slot, sync_height_rx) =
             Worker::new(storage.clone(), node.clone()).await?;
 
-        tokio::spawn(worker.run());
+        let worker_handle = tokio::spawn(worker.run().map(|_| ()));
 
         Ok(Self {
             storage,
             error_slot,
             sync_height_rx,
             state_commitment_tree: sct,
             node,
+            worker_handle: Arc::new(worker_handle),
         })
     }
 
+    /// Close this view service, stopping the background syncing task.
+    ///
+    /// This will have a spoooky action at a distance in that other cloned instances of this
+    /// service sharing the syncing task will also be silently broken, so be careful when using
+    /// this method.
+    ///
+    /// Also this is sort of a stop-gap for a more robust view service where this method won't be
+    /// necessary, so one should exercise skepticism that this method even exists.
+    pub async fn abort(&self) {
+        self.worker_handle.abort();
```

Comment:
Random observation, but if you want some stronger guarantees about the timing of the join handle cancellation, you can immediately await the worker handle and that will make this method return when those resources are actually released. It can be useful for testing.